x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -0,0 +1,294 @@
+      },
+    ],
+    engines: [
+      [],
+      {
+        setCredentialsDetails: (_, { engines }) => engines,
+      },
+    ],
+    nameInputBlurred: [
+      false,
+      {
+        setNameInputBlurred: (_, nameInputBlurred) => nameInputBlurred,
+      },
+    ],
+    apiUrl: [
+      '',
+      {
+        setCredentialsDetails: (_, { apiUrl }) => apiUrl,
+      },
+    ],
+    activeApiToken: [
+      defaultApiToken,
+      {
+        addEngineName: (activeApiToken, engineName) => ({
+          ...activeApiToken,
+          engines: [...(activeApiToken.engines || []), engineName],
+        }),
+        removeEngineName: (activeApiToken, engineName) => ({
+          ...activeApiToken,
+          engines: (activeApiToken.engines || []).filter((name) => name !== engineName),
+        }),
+        setAccessAllEngines: (activeApiToken, accessAll) => ({
+          ...activeApiToken,
+          access_all_engines: accessAll,
+          engines: accessAll ? [] : activeApiToken.engines,
+        }),
+        onApiTokenCreateSuccess: () => defaultApiToken,
+        onApiTokenUpdateSuccess: () => defaultApiToken,
+        setTokenName: (activeApiToken, name) => ({ ...activeApiToken, name: formatApiName(name) }),
+        setTokenReadWrite: (activeApiToken, { name, checked }) => ({
+          ...activeApiToken,
+          [name]: checked,
+        }),
+        setTokenType: (activeApiToken, tokenType) => ({
+          ...activeApiToken,
+          access_all_engines: tokenType === ADMIN ? false : activeApiToken.access_all_engines,
+          engines: tokenType === ADMIN ? [] : activeApiToken.engines,
+          write: tokenType === PRIVATE,
+          read: tokenType === PRIVATE,
+          type: tokenType,
+        }),
+        toggleCredentialsForm: (_, activeApiToken) => activeApiToken || defaultApiToken,
+      },
+    ],
+    activeApiTokenRawName: [
+      '',
+      {
+        setTokenName: (_, activeApiTokenRawName) => activeApiTokenRawName,
+        toggleCredentialsForm: (activeApiTokenRawName, activeApiToken) =>
+          activeApiToken.name || activeApiTokenRawName,
+        hideCredentialsForm: () => '',
+        onApiTokenCreateSuccess: () => '',
+        onApiTokenUpdateSuccess: () => '',
+      },
+    ],
+    activeApiTokenIsExisting: [
+      false,
+      {
+        toggleCredentialsForm: (_, activeApiToken) => !!activeApiToken.id,
+      },
+    ],
+    showCredentialsForm: [
+      false,
+      {
+        toggleCredentialsForm: (showCredentialsForm) => !showCredentialsForm,
+        hideCredentialsForm: () => false,
+        onApiTokenCreateSuccess: () => false,
+        onApiTokenUpdateSuccess: () => false,
+      },
+    ],
+    formErrors: [
+      [],
+      {
+        onApiTokenError: (_, formErrors) => formErrors,
+        onApiTokenCreateSuccess: () => [],
+        toggleCredentialsForm: () => [],
+        resetCredentials: () => [],
+      },
+    ],
+    // TODO flashMessages from ent-search
```

Comment:
See above, you can delete this line

Reply:
b6d0fb572072f189c0dd22a607a051fc7ec4b546